packages/rocketchat-ui-flextab/client/tabs/userActions.js

```diff
@@ -55,7 +55,8 @@ export const getActions = function({ user, directActions, hideAdminControls }) {
 	};
 	const canDirectMessage = (username) => {
 		const user = Meteor.user();
-		return RocketChat.authz.hasAllPermission('create-d') && user && user.username !== username;
+		const rid = Session.get('openedRoom');
+		return username && (!directActions && rid && RocketChat.models.Subscriptions.findOne({ rid }).name !== username) || (directActions && RocketChat.authz.hasAllPermission('create-d') && user && user.username !== username);
```

Comment:
what about describing what each conditions does assigning a new variable?

```
const whatThisDoes = (!directActions && rid && RocketChat.models.Subscriptions.findOne({ rid }).name !== username);
const whatThisDoesAsWell = (directActions && RocketChat.authz.hasAllPermission('create-d') && user && user.username !== username);

return username && whatThisDoes || whatThisDoesAsWell;
```

this will make the code easy to read and understand